Examples/SampleCat/SampleCat/UserViewModel.swift

```diff
@@ -0,0 +1,82 @@
+import Foundation
+import RevenueCat
+import SwiftUI
+
+@MainActor @Observable final class UserViewModel {
+    /* The latest CustomerInfo from RevenueCat. Updated by the `customerInfoStream` in the initializer. */
+    var customerInfo: CustomerInfo? {
+        didSet {
+            guard let entitlementIdentifier = Constants.entitlementIdentifier else { return }
+            subscriptionActive = customerInfo?.entitlements[entitlementIdentifier]?.isActive == true
+        }
+    }
+
+    /* The latest offerings */
+    var offerings: Offerings?
+
+    /* Checks if a subscription is active for a given entitlement */
+    var subscriptionActive: Bool = false
+
+    var isFetchingOfferings: Bool = false
+
+    var isPurchasing: Bool = false
+
+    init() {
+        // Configure the SDK with the API Key
+        Purchases.configure(withAPIKey: Constants.apiKey)
+        /* Listen to changes in the `customerInfo` object using an `AsyncStream` */
+        Task {
+            for await newCustomerInfo in Purchases.shared.customerInfoStream {
+                await MainActor.run { customerInfo = newCustomerInfo }
+            }
+        }
+    }
+
+    func purchase(_ product: StoreProduct) async {
+        isPurchasing = true
+        defer { isPurchasing = false }
+
+        do {
+            let (_, customerInfo, userCancelled) = try await Purchases.shared.purchase(product: product)
```

Comment:
Should we try to move devs to use `purchase(package:)` rather than this one? We potentially miss a lot of data if the developer doesn't get the product from an offering